File: apps/movil/lib/pantallas/soporte/catalogo/primeros_pasos.dart
import '../../../dominio/tutoriales/modelo.dart';

/// La categoría con la que se agrupan en el centro de ayuda.
const String primerosPasos = 'Primeros pasos';

/// Los tutoriales que le sirven a cualquiera, tenga o no un pasanaku andando.
/// Son DATOS: ninguno importa un widget ni sabe cómo se dibuja un globo.
const TutorialDefinicion introALaApp = TutorialDefinicion(
  id: 'intro-app',
  version: '1.0.0',
  titulo: 'Qué es AportaYa',
  descripcion:
      'El pasanaku de siempre, sin el cuaderno. Qué ves en la pantalla de inicio y qué podés hacer.',
  categoria: primerosPasos,
  ruta: '/billetera/inicio',
  minutos: 3,
  dificultad: Dificultad.inicial,
  obligatorio: true,
  siguiente: 'navegacion-app',
  pasos: [
    PasoDeTutorial(
      id: 'saldo',
      titulo: 'Acá está tu plata',
      descripcion:
          'Este es el saldo de tu billetera. Lo que ves es lo que hay: AportaYa no cobra nada que no te haya mostrado antes.',
      ancla: 'billetera.saldo',
      ruta: '/billetera/inicio',
    ),
    PasoDeTutorial(
      id: 'acciones',
      titulo: 'Tres cosas para hacer',
      descripcion:
          'Aportar es lo que mantiene vivo tu pasanaku, por eso es el botón naranja. Recargar mete plata a la billetera; retirar la saca.',
      ancla: 'billetera.acciones',
      posicion: PosicionDeGlobo.arriba,
    ),
    PasoDeTutorial(
      id: 'pestanas',
      titulo: 'Y todo lo demás vive acá abajo',
      descripcion:
          'Inicio, Grupos, Movimientos, Perfil y Ayuda. Siempre están: no hace falta volver atrás para cambiar de lugar.',
      ancla: 'shell.pestanas',
      posicion: PosicionDeGlobo.arriba,
    ),
  ],
);

const TutorialDefinicion navegacionDeLaApp = TutorialDefinicion(
  id: 'navegacion-app',
  version: '1.0.0',
  titulo: 'Moverse por la app',
  descripcion: 'Qué hay en cada pestaña y cómo volver a donde estabas.',
  categoria: primerosPasos,
  ruta: '/billetera/inicio',
  minutos: 2,
  dificultad: Dificultad.inicial,
  requisitos: ['intro-app'],
  siguiente: 'usar-ayuda',
  pasos: [
    PasoDeTutorial(
      id: 'barra',
      titulo: 'La barra de abajo es el mapa',
      descripcion:
          'Cada pestaña recuerda dónde estabas: si salís de Grupos y volvés, volvés al mismo lugar.',
      ancla: 'shell.pestanas',
      posicion: PosicionDeGlobo.arriba,
      ruta: '/billetera/inicio',
    ),
    PasoDeTutorial(
      id: 'grupos',
      titulo: 'Probá ir a Grupos',
      descripcion:
          'Ahí está el estado de tus pasanakus: qué te toca aportar y cuándo es tu turno. Tocá «Grupos» para seguir.',
      ancla: 'shell.pestanas',
      posicion: PosicionDeGlobo.arriba,
      accion: AccionEsperada.navegar('/pasanaku/mi-estado'),
      ayudaSiFalla: 'Tocá «Grupos» en la barra de abajo para seguir.',
    ),
    PasoDeTutorial(
      id: 'volver',
      titulo: 'Y volver siempre es posible',
      descripcion:
          'Tocás «Inicio» y estás de nuevo en tu saldo. Nada de lo que estabas mirando se pierde.',
      ancla: 'pasanaku.mi-estado',
      ruta: '/pasanaku/mi-estado',
    ),
  ],
);

const TutorialDefinicion usarElCentroDeAyuda = TutorialDefinicion(
  id: 'usar-ayuda',
  version: '1.0.0',
  titulo: 'Usar el centro de ayuda',
  descripcion:
      'Buscar un tutorial, ver cuánto llevás hecho y repetir el que quieras.',
  categoria: primerosPasos,
  ruta: '/soporte/ayuda',
  minutos: 2,
  dificultad: Dificultad.inicial,
  pasos: [
    PasoDeTutorial(
      id: 'avance',
      titulo: 'Tu avance, de un vistazo',
      descripcion: 'Cuántos tutoriales completaste de los que hay para vos.',
      ancla: 'ayuda.avance',
      ruta: '/soporte/ayuda',
    ),
    PasoDeTutorial(
      id: 'buscar',
      titulo: 'Buscá lo que necesitás',
      descripcion:
          'Escribí «aportar», «turno», «retirar». La lista se recorta mientras escribís.',
      ancla: 'ayuda.buscador',
    ),
    PasoDeTutorial(
      id: 'repetir',
      titulo: 'Ninguno se gasta',
      descripcion:
          'Podés repetir cualquier tutorial las veces que quieras. «Reiniciar» lo deja como el primer día.',
      ancla: 'ayuda.lista',
      posicion: PosicionDeGlobo.arriba,
    ),
  ],
);

File: apps/backoffice/src/app/rutas/ayuda/catalogo/primeros-pasos.ts
import type { TutorialDefinicion } from '@aportaya/tutoriales/tipos'

/** La categoría con la que se agrupan en el centro de ayuda. */
export const PRIMEROS_PASOS = 'Primeros pasos'

/**
 * Los cuatro tutoriales que puede hacer cualquiera con sesión abierta, sin importar su
 * rol. Son DATOS: ninguno importa un componente ni sabe cómo se dibuja un globo.
 */
export const introduccionALaPlataforma: TutorialDefinicion = {
  id: 'intro-plataforma',
  version: '1.0.0',
  titulo: 'Qué es el backoffice de AportaYa',
  descripcion: 'De qué se ocupa esta herramienta, qué parte te toca a vos y dónde está cada cosa.',
  categoria: PRIMEROS_PASOS,
  ruta: '/tablero',
  minutos: 3,
  dificultad: 'inicial',
  obligatorio: true,
  siguiente: 'navegacion-principal',
  pasos: [
    {
      id: 'bienvenida',
      titulo: 'Esto es el backoffice',
      descripcion: 'Desde acá se atiende lo que la app del participante no resuelve sola: verificar identidades, mirar saldos, cerrar períodos, revisar campañas. Ninguna pantalla de acá la ve un participante.',
      objetivo: 'tablero-accesos',
      posicion: 'abajo',
      automatica: 'desplazar',
    },
    {
      id: 'tu-rol',
      titulo: 'Tu rol decide qué ves',
      descripcion: 'Acá arriba está el rol con el que entraste. El menú muestra solo las secciones que tu rol abre; si no ves una, no es un error.',
      objetivo: 'cabecera-rol',
      posicion: 'abajo',
    },
    {
      id: 'el-menu',
      titulo: 'El menú es el mapa',
      descripcion: 'Cada entrada es un dominio completo. Se puede volver al tablero en cualquier momento sin perder nada de lo que estabas mirando.',
      objetivo: 'menu-principal',
      posicion: 'derecha',
    },
    {
      id: 'ayuda-en-pantalla',
      titulo: 'Este botón te trae de vuelta',
      descripcion: 'En cualquier pantalla con tutorial vas a ver este botón. Lo pulsás y te explica esa pantalla, sin salir de ella.',
      objetivo: 'lanzador-de-tutorial',
      posicion: 'abajo',
    },
  ],
}

export const navegacionPrincipal: TutorialDefinicion = {
  id: 'navegacion-principal',
  version: '1.0.0',
  titulo: 'Moverse por las secciones',
  descripcion: 'Cómo ir de una sección a otra, qué hay en cada una y cómo volver.',
  categoria: PRIMEROS_PASOS,
  ruta: '/tablero',
  minutos: 3,
  dificultad: 'inicial',
  requisitos: ['intro-plataforma'],
  siguiente: 'usar-el-centro-de-ayuda',
  pasos: [
    {
      id: 'tablero',
      titulo: 'El tablero es el punto de entrada',
      descripcion: 'Cada tarjeta lleva a un dominio y dice de qué se ocupa. Es lo que ves al entrar, y a lo que volvés con el logo.',
      objetivo: 'tablero-accesos',
      ruta: '/tablero',
      posicion: 'abajo',
    },
    {
      id: 'abrir-una-seccion',
      titulo: 'Probá abrir una sección',
      descripcion: 'Pulsá cualquier entrada del menú. Te llevamos con vos: el tutorial sigue abierto del otro lado.',
      objetivo: 'menu-principal',
      posicion: 'derecha',
      accion: { tipo: 'clic', objetivo: 'menu-principal' },
      ayudaSiFalla: 'Pulsá una de las entradas del menú de la izquierda para seguir.',
    },
    {
      id: 'volver',
      titulo: 'Y volver siempre es posible',
      descripcion: 'El menú no se va nunca: está en todas las pantallas, con la sección actual marcada. No hace falta el botón de atrás del navegador.',
      objetivo: 'menu-principal',
      posicion: 'derecha',
    },
  ],
}

export const tuSesionYTuRol: TutorialDefinicion = {
  id: 'sesion-y-rol',
  version: '1.0.0',
  titulo: 'Tu sesión, tu rol y lo que podés hacer',
  descripcion: 'Por qué dos personas ven cosas distintas y qué significa que una acción esté deshabilitada.',
  categoria: PRIMEROS_PASOS,
  ruta: '/tablero',
  minutos: 2,
  dificultad: 'inicial',
  requisitos: ['intro-plataforma'],
  pasos: [
    {
      id: 'el-rol',
      titulo: 'Entraste con un rol',
      descripcion: 'El rol es lo que el servidor usa para decidir qué te deja hacer. No se cambia desde acá: lo asigna Sistemas.',
      objetivo: 'cabecera-rol',
      posicion: 'abajo',
    },
    {
      id: 'el-menu-filtrado',
      titulo: 'El menú ya viene filtrado',
      descripcion: 'Solo aparecen las secciones que tu rol abre. Ocultar es una comodidad: aunque escribieras la dirección a mano, el servidor te diría que no.',
      objetivo: 'menu-principal',
      posicion: 'derecha',
    },
    {
      id: 'acciones-apagadas',
      titulo: 'Una acción apagada no es una falla',
      descripcion: 'Cuando un botón está deshabilitado es porque falta algo —un dato, una firma, un permiso—. La pantalla siempre dice qué falta; si no lo dice, es un error que vale la pena reportar.',
      objetivo: 'tablero-accesos',
      posicion: 'abajo',
    },
  ],
}

export const usarElCentroDeAyuda: TutorialDefinicion = {
  id: 'usar-el-centro-de-ayuda',
  version: '1.0.0',
  titulo: 'Usar el centro de tutoriales',
  descripcion: 'Buscar, filtrar, retomar lo que dejaste a medias y repetir lo que quieras.',
  categoria: PRIMEROS_PASOS,
  ruta: '/ayuda',
  minutos: 2,
  dificultad: 'inicial',
  pasos: [
    {
      id: 'avance',
      titulo: 'Tu avance, de un vistazo',
      descripcion: 'Acá está cuánto llevás hecho. Cuenta los tutoriales completados sobre los que tu rol puede ver.',
      objetivo: 'ayuda-avance',
      ruta: '/ayuda',
      posicion: 'abajo',
    },
    {
      id: 'buscar',
      titulo: 'Buscar por palabra',
      descripcion: 'Escribí lo que querés aprender —«campaña», «expediente», «período»— y la lista se recorta mientras escribís.',
      objetivo: 'ayuda-buscador',
      posicion: 'abajo',
      accion: { tipo: 'escribir', objetivo: 'ayuda-buscador', minimo: 2 },
      ayudaSiFalla: 'Escribí al menos dos letras en el buscador para seguir.',
      automatica: 'enfocar',
    },
    {
      id: 'filtrar',
      titulo: 'Filtrar por estado y categoría',
      descripcion: 'Pendientes, en progreso, completados u obligatorios. Es la forma rápida de ver qué te falta.',
      objetivo: 'ayuda-filtros',
      posicion: 'abajo',
    },
    {
      id: 'repetir',
      titulo: 'Repetir cuesta un clic',
      descripcion: 'Ningún tutorial se agota. «Reiniciar» borra tu avance de ese tutorial y lo deja como el primer día.',
      objetivo: 'ayuda-lista',
      posicion: 'arriba',
    },
  ],
}
